mlir/lib/Transforms/Utils/Inliner.cpp

```diff
@@ -344,6 +344,28 @@ static void collectCallOps(iterator_range<Region::iterator> blocks,
     }
   }
 }
+//===----------------------------------------------------------------------===//
+// Inliner
+//===----------------------------------------------------------------------===//
+// Initialize doClone function with the default implementation
+Inliner::CloneCallbackTy &Inliner::doClone() {
+  static Inliner::CloneCallbackTy doWork =
```

Comment:
This can't work: you're using basically a single global variable that will be used for the entire process, shared by all inliner instance. That would break many use-cases (and isn't thread-safe).